frugalos_segment/src/client/storage.rs

```diff
@@ -340,12 +342,16 @@ impl DispersedClient {
         }
     }
     pub fn get_fragment(self, local_node: NodeId, version: ObjectVersion) -> GetDispersedFragment {
-        let mut spares = self
+        let candidates = self
             .cluster
             .candidates(version)
-            .filter(|m| m.node != local_node)
             .cloned()
             .collect::<Vec<_>>();
+        // we don't return Result here because using Result in this context makes the code too much complicated.
+        let participants = Participants::dispersed(&candidates, self.config.fragments())
+            .expect(&format!("Found a bug: the configuration must be validated before use. fragments(len = {}) must be lower than candidates(len = {}", self.config.fragments(), candidates.len()));
```

Comment:
ここで `Participants::dispersed` が `Err` を返してしまわないように Erasure Code の設定をしている場所で最低限 valid な状態を保証して欲しいという気持ちです。

Reply:
Let me think about this...

Reply:
Yes, I know this is a controversial decision.

Reply:
I think `self.cluster` reflects the initial configuration when building a system (and does not change dynamically) because the type of `self.cluster` is `ClusterConfig`.

So I **believe** that checking `self.cluster.members.len() >= self.config.fragments()` is performed immediately after a user builds a system but I can not find such a code now 😢

Reply:
Finally, we remove this validation and resolves the validation issue in another PR.
issue: https://github.com/frugalos/frugalos/issues/35